src/execution/experimental-executions.js

```diff
@@ -311,53 +464,60 @@ const createDerivedMetric = (measure, mdObj, measureIndex) => {
     return {
         element,
         definition,
-        sort: getMetricSort(sort),
+        sort,
         meta: {
             measureIndex
         }
     };
 };
 
-const createContributionMetric = (measure, mdObj, measureIndex) => {
-    const category = first(getCategories(mdObj));
-    const getMetricExpression = partial(getPercentMetricExpression, category);
+const createContributionMetric = (measure, mdObj, measureIndex, attributesMap) => {
+    const category = first(getCategories(getBuckets(mdObj)));
+    const getMetricExpression = partial(getPercentMetricExpression, category, attributesMap);
     const title = getBaseMetricTitle(get(measure, 'title'));
     const hasher = partial(getGeneratedMetricHash, title, CONTRIBUTION_METRIC_FORMAT);
+    const identifier = getGeneratedMetricIdentifier(measure, 'percent', getMetricExpression, hasher, attributesMap);
     return {
-        element: getGeneratedMetricIdentifier(measure, 'percent', getMetricExpression, hasher),
+        element: identifier,
         definition: {
             metricDefinition: {
-                identifier: getGeneratedMetricIdentifier(measure, 'percent', getMetricExpression, hasher),
-                expression: getMetricExpression(measure),
+                identifier,
+                expression: getMetricExpression(measure, attributesMap),
                 title,
                 format: CONTRIBUTION_METRIC_FORMAT
             }
         },
-        sort: getMetricSort(get(measure, 'sort')),
+        sort: getMeasureSorting(measure, mdObj),
         meta: {
             measureIndex
         }
     };
 };
 
-const createPoPMetric = (measure, mdObj, measureIndex) => {
-    const title = getPoPMetricTitle(get(measure, 'title'));
-    const format = get(measure, 'format');
+function getOriginalMeasureForPoP(popMeasure, mdObj) {
+    return getMeasures(getBuckets(mdObj)).find(measure =>
+        get(measure, 'localIdentifier') === get(getPoPDefinition(popMeasure), ['measureIdentifier'])
+    );
+}
+
+const createPoPMetric = (popMeasure, mdObj, measureIndex, attributesMap) => {
+    const title = getBaseMetricTitle(get(popMeasure, 'title'));
+    const format = get(popMeasure, 'format');
     const hasher = partial(getGeneratedMetricHash, title, format);
 
-    const date = getDate(mdObj);
+    const attributeUri = get(popMeasure, 'definition.popMeasureDefinition.popAttribute.uri');
+    const originalMeasure = getOriginalMeasureForPoP(popMeasure, mdObj);
 
-    let generated;
-    let getMetricExpression = partial(getPoPExpression, date, `[${get(measure, 'objectUri')}]`);
+    let getMetricExpression = partial(getPoPExpression, attributeUri, `[${get(getDefinition(originalMeasure), ['item', 'uri'])}]`);
```

Comment:
This is quite unreadable -> expand to more lines?

Reply:
fixed